ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -77,6 +80,34 @@
     return Optional.of(result);
   }
 
+  private static Optional<?> coerceStruct(Object value, SqlStruct targetType) {
+    if (!(value instanceof Struct)) {
+      return Optional.empty();
+    }
+
+    final Struct struct = (Struct) value;
+    final Struct coerced = new Struct(
+        SchemaConverters.sqlToConnectConverter().toConnectSchema(targetType)
+    );
+
+    for (Field field : coerced.schema().fields()) {
+      Optional<io.confluent.ksql.schema.ksql.types.Field> sqlField = targetType.field(field.name());
+      if (!sqlField.isPresent()) {
+        // if there was a field in the struct that wasn't in the schema
+        // we should throw an exception
```

Comment:
nit: we don't know the caller will throw an exception on `Optional.empty`.  I'd just say 'if there is a field in the struct that isn't in the schema we can not coerce'